### PR TITLE
feat: add iconify logos

### PR DESCRIPTION
**Pull request by hiroto1220**
The architecture diagram is now available in mermaid v11.1.0.
However, currently iconify logos cannot be seen in VSCode.
This pull request will allow iconify logos to be used.

Before
スクリーンショット 2024-09-08 1 22 13

after
スクリーンショット 2024-09-08 1 19 40

### CODE REVIEW

File: markdownPreview/index.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-function init() {
+function init() { 
     const configSpan = document.getElementById('markdown-mermaid');
     const darkModeTheme = configSpan?.dataset.darkModeTheme;
     const lightModeTheme = configSpan?.dataset.lightModeTheme;
+  
+    mermaid.registerIconPacks([
+      {
+        name: 'logos',
+        loader: () => Promise.resolve(logos.icons),
```
Comment (mjbvz) [inline markdownPreview/index.ts:13]: Can this be a lazy import so it's unload loaded if needed?

File: markdownPreview/index.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 import { renderMermaidBlocksInElement } from './mermaid';
+import * as logos from '@iconify-json/logos';
 
-function init() {
+function init() { 
     const configSpan = document.getElementById('markdown-mermaid');
     const darkModeTheme = configSpan?.dataset.darkModeTheme;
     const lightModeTheme = configSpan?.dataset.lightModeTheme;
+  
+    mermaid.registerIconPacks([
```
Comment (mjbvz) [inline markdownPreview/index.ts:10]: This should likely happen in shared code, see `extendMarkdownItWithMermaid`. That way it works both in notebooks and the markdown previews

Comment (hiroto1220) [reply markdownPreview/index.ts]: I have created `registerIconPacks` in `src/mermaid.ts `.

File: markdownPreview/index.ts (hunk 1 above)
Comment (hiroto1220) [reply markdownPreview/index.ts]: Yes! 
I have changed to lazy import !

**Comment by HannesOberreiter** [comment]
Just saw the pull request as I also started to use the architecture feature. Actually in my opinion the and link to icon set should be a config, otherwise where to start and end. 

eg. `markdown-mermaid.icon-packs`

```
Array<{
 prefix?: string,
 pack: string
}>
```

**Comment by hiroto1220** [comment]
@HannesOberreiter 
I have implemented a fix based on your feedback.
Thank you for the valuable suggestion!

**Comment by HannesOberreiter** [comment]
I did not meant to simply extend the list of lazy load icons. I meant a new VSCode config for this extension, so each user can define it owns icon sets he wants to be loaded. As there over 100 icon sets to choose from.

**Comment by hiroto1220** [comment]
That makes sence.  VSCode extension is not intended for use in a local environment, wouldn't it be unnecessary to have files for individual users to load their own icon sets?

**Comment by HannesOberreiter** [comment]
> VSCode extension is not intended for use in a local environment, wouldn't it be unnecessary to have files for individual users to load their own icon sets?

If you mean that settings might differ between VSCode users working on the same project or repository, I don't see an issue. You can configure these settings in a local `.vscode` folder.

As I mentioned previously, in my opinion the icon sets should be a config. Therefore we need to add the config to the `package.json`. Load the user selected icons in the root extension `activate` function.

I never did program an VSCode extension but I believe a refactor would be helpful to add mermaid as extension to `markdown-it` and not load it on its own on preview open. Then one could lazy load the icon set in the main `activate` function. Otherwise some html tricks are necessary to get the user config forwarded to the preview window (only looked quickly so there may be other ways).

PS: A completely different approach is done by `vscode-markdown-preview-enhanced` if you want to load icon sets, where you can define your own html header and load the icons in the header. But I do not like this approach as it is no config anymore and must set it for each user.

Cheers
Hannes

**Comment by mjbvz** [comment]
Thanks! Looks good in my testing. Will publish shortly in 1.26

**Pull request merged**